Authorization

Wahaj-Ali opened the pull request:
## Highlights
- Limit access to web app resources based on authorization rules.
- Install CanCanCan.
- Added a role column to the users' table. Used a migration for this.
- Only authorized users can see the Delete button for posts and comments.
- A user can delete only his posts and comments.
- An admin can delete all posts and comments

Wahaj-Ali committed:
Install CanCanCan

Wahaj-Ali committed:
Add a column 'role' in the users table using migration.

Wahaj-Ali committed:
Create ability.rb

Wahaj-Ali committed:
Update ability.rb

A user can delete a post if it is theirs or if they have an admin role.

Wahaj-Ali committed:
Update posts_controller.rb

Add destroy method to delete a post.

Wahaj-Ali committed:
Update post.rb

Add dependence of comments to the post in order to delete a post with comments without any error.

Wahaj-Ali committed:
Update routes.rb

Add routes to the destroy method.

Wahaj-Ali committed:
Update show.html.erb

Add a delete button to the posts of the current user only.

Wahaj-Ali committed:
Remove linter errors.

Wahaj-Ali committed:
Update show.html.erb

Add a delete comment button shown only to the author of comments and the admins.

Wahaj-Ali committed:
Update application.css

Add style to the delete comment button.

Ismailco commented on app/controllers/posts_controller.rb:
```diff
@@ -28,4 +28,18 @@ def create
       render :new
     end
   end
+
```
- [x] You should use `CanCanCan` for Post authorization in `Post Controller`, kindly make sure to implement this. you can add this line to your controller ``load_and_authorize_resource``

Wahaj-Ali replied:
Thank you @Ismailco for pointing out the issue. I have resolved it 😄

Ismailco reviewed (changes requested):
### Hi @Wahaj-Ali, @codehass, 

Good job so far 👍🏼 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

codehass committed:
Add destroy method for comment controller

Wahaj-Ali committed:
Made required changes.

In this commit
- use CanCanCan for Post authorization in Post Controller by using load_and_authorize_resource.

Wahaj-Ali committed:
Update posts_controller.rb

Remove linter errors.

Owusu-Desmond reviewed (approved):
Hi @Wahaj-Ali and @Ismailco , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

<img src="https://media4.giphy.com/media/3kuSo744UIPJjcJUEn/giphy.gif"/>

## Highlights
📌 No Linters error ✅
📌 Descriptive pull request✅
📌 Good commit messages ✅
📌 Professional README file ✅

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

codehass commented:
Hi @Owusu-Desmond thank you for your review, you approved the project in my dashboard can you cancel it, we should submit it individually.
Thank you 😁
